packages/calling-stateful-client/src/CallAgentDeclarative.ts

```diff
@@ -104,18 +131,20 @@ class ProxyCallAgent implements ProxyHandler<CallAgent> {
       incomingCallSubscriber.unsubscribe();
       this._incomingCallSubscribers.delete(incomingCallId);
     }
+    this._declarativeIncomingCalls.delete(incomingCallId);
     this._context.setIncomingCallEnded(incomingCallId, callEndReason);
   };
 
-  private incomingCall = (event: { incomingCall: IncomingCall }): void => {
+  private incomingCall: IncomingCallEvent = ({ incomingCall }): void => {
     // Make sure to not subscribe to the incoming call if we are already subscribed to it.
-    if (!this._incomingCallSubscribers.has(event.incomingCall.id)) {
+    if (!this._incomingCallSubscribers.has(incomingCall.id)) {
       this._incomingCallSubscribers.set(
-        event.incomingCall.id,
-        new IncomingCallSubscriber(event.incomingCall, this.setIncomingCallEnded)
+        incomingCall.id,
+        new IncomingCallSubscriber(incomingCall, this.setIncomingCallEnded)
       );
     }
-    this._context.setIncomingCall(convertSdkIncomingCallToDeclarativeIncomingCall(event.incomingCall));
+    this._declarativeIncomingCalls.set(incomingCall.id, incomingCallDeclaratify(incomingCall, this._context));
```

Comment:
@PorterNan Can te `Call.id` change for an incoming call (before it is accepted / rejected)? If yes, we might need to handle the `idChanged` event similar to ongoing call.

Reply:
Haha, confirmed with Artur. It doesn't change :)